spec/requests/saved_pages_request_spec.rb

```diff
@@ -0,0 +1,169 @@
+        expect(error["type"]).to eq(I18n.t("errors.cannot_save_page.type"))
+        expect(error["detail"]).to eq(I18n.t("errors.cannot_save_page.detail", page_path: ""))
+        expect(error["page_path"]).to eq("")
+        expect(error["errors"]["page_path"]).to include("can't be blank")
+      end
+
+      it "returns status 422 Unprocessable Entity with a RFC 7807 'problem detail' object if the page path is missing" do
+        post saved_pages_path, headers: headers
+
+        expect(response).to have_http_status(:unprocessable_entity)
+        error = JSON.parse(response.body)
+        expect(error["title"]).to eq(I18n.t("errors.cannot_save_page.title"))
+        expect(error["type"]).to eq(I18n.t("errors.cannot_save_page.type"))
+        expect(error["detail"]).to eq(I18n.t("errors.cannot_save_page.detail", page_path: nil))
+        expect(error["page_path"]).to eq(nil)
+        expect(error["errors"]["page_path"]).to include("can't be blank")
+      end
+
+      it "returns status 422 Unprocessable Entity with a RFC 7807 'problem detail' object if the page path contains query parameters" do
+        page_path = "/foo/bar?uhoh"
+        post saved_pages_path, headers: headers, params: { page_path: page_path }.to_json
+
+        expect(response).to have_http_status(:unprocessable_entity)
+        error = JSON.parse(response.body)
+        expect(error["title"]).to eq(I18n.t("errors.cannot_save_page.title"))
+        expect(error["type"]).to eq(I18n.t("errors.cannot_save_page.type"))
+        expect(error["detail"]).to eq(I18n.t("errors.cannot_save_page.detail", page_path: page_path))
+        expect(error["page_path"]).to eq(page_path)
+        expect(error["errors"]["page_path"]).to include("must only include URL path")
+      end
+
+      it "returns status 422 Unprocessable Entity with a RFC 7807 'problem detail' object if the page path contains a fragment identifier" do
+        page_path = "/foo/bar#heading1"
+        post saved_pages_path, headers: headers, params: { page_path: page_path }.to_json
+
+        expect(response).to have_http_status(:unprocessable_entity)
+        error = JSON.parse(response.body)
+        expect(error["title"]).to eq(I18n.t("errors.cannot_save_page.title"))
+        expect(error["type"]).to eq(I18n.t("errors.cannot_save_page.type"))
+        expect(error["detail"]).to eq(I18n.t("errors.cannot_save_page.detail", page_path: page_path))
+        expect(error["page_path"]).to eq(page_path)
+        expect(error["errors"]["page_path"]).to include("must only include URL path")
+      end
+    end
+
+    describe "DELETE /api/saved_pages" do
+      it "returns status 204 No Content if a record has been sucessfully deleted" do
+        FactoryBot.create(:saved_page, oidc_user_id: user.id, page_path: "/page-path/1")
+        delete saved_page_path(page_path: "/page-path/1"), headers: headers
+
+        expect(response).to have_http_status(:no_content)
+      end
```

Comment:
We're not testing that a record gets deleted.

Reply:
[Done](https://github.com/alphagov/account-api/pull/69/commits/ed5f50b71aa55fa860315f20d0802db74e244b34#diff-e494be0ec09eb92d5782bcb4d976f0a1c24e9f3874d5b2a1e980ac902d51eda7R127-R133)